label-application-rule-config-service-api/src/main/proto/org/hypertrace/label/application/rule/config/service/v1/label_application_rule.proto

```diff
@@ -0,0 +1,104 @@
+syntax = "proto3";
+
+package org.hypertrace.label.application.rule.config.service.v1;
+
+option java_multiple_files = true;
+
+message LabelApplicationRule {
+  string id = 1;
+  LabelApplicationRuleData data = 2;
+}
+
+message LabelApplicationRuleData {
+  string name = 1;
+  MatchingCriteria matching_criteria = 2;
+  ApplicationCriteria application_criteria = 3;
+
+  message MatchingCriteria {
+    Condition condition = 1;
```

Comment:
If we're down to a single field in this, we can probably drop a layer of envelopes.

Reply:
Condition is used recursively in CompositeCondition. Let's keep matching and application criteria?

Reply:
What about changing line 14 to `Condition matching_criteria = 2;` That shouldn't get in the way of the recursion. I'm fine either way though. More than anything, I just want application criteria to be renamed since it _isn't_ a criteria.

Reply:
yea Condition matching_criteria is more readable but ApplicationCriteria is confusing.A rule essentially has conditions and actions so may be Actions ? ( I see Action is already used)

Reply:
Removed both criteria, and used condition and action directly.